src/console/TestController.php

```diff
@@ -61,7 +61,19 @@ protected function runInit() {
      * Run the tests
      */
     protected function runTests() {
-        $process = new Process(['./vendor/bin/pest']);
+        $params = $this->request->getParams();
+        $pestOptions = [];
+        $stdOutIndex = array_search('--', $params, true);
+
+        if ($stdOutIndex !== false) {
+            $pestOptions = array_slice($params, ++$stdOutIndex);
+
+            if (array_search('--coverage', $pestOptions, true) !== false) {
+                $_ENV['XDEBUG_MODE'] = 'coverage';
+            }
```

Comment:
This is the only bits that make me nervous because there's no way for a user to override it. `--coverage` will _force_ `XDEBUG_MODE`.

It should be possible to `XDEBUG_MODE=coverage ./craft pest/test`, right? But we just need to pass the `ENV` through from Craft to Pest?

Reply:
One last thought, while we're on this train. Sometimes I like to run `php -d "xdebug.mode=coverage"` to pass in vars… Or, maybe more likely is something like `php -d "memory_limit=-1"` Is it feasible to account for passing stuff like that through? E.g., if I ran,

```shell
$ php -d "xdebug.mode=coverage" craft pest/test
```

Again, just making sure we're covering all options here

Reply:
no not at all! That makes sense. i've never actually used `pcov` before for tests. Now that you mention that I see pest supports that and `phpdbg` as well as xdebug.

I can remove this bit for now and maybe revist? It's not the end of world if people have to come up with their own logic to set an env variable for coverage.